src/frontend/packages/lib_video/src/components/common/VideoWidgetProvider/widgets/LicenseManager/index.spec.tsx

```diff
@@ -0,0 +1,118 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import fetchMock from 'fetch-mock';
+import { useJwt, videoMockFactory } from 'lib-components';
+import { render } from 'lib-tests';
+import React from 'react';
+
+import { InfoWidgetModalProvider } from 'hooks/useInfoWidgetModal';
+import { wrapInVideo } from 'utils/wrapInVideo';
+
+import { LicenseManager } from '.';
+
+jest.mock('lib-components', () => ({
+  ...jest.requireActual('lib-components'),
+  report: jest.fn(),
+}));
+
+const licenseChoices = [
+  { display_name: 'Creative Common By Attribution', value: 'CC_BY' },
+  { display_name: 'All rights reserved', value: 'NO_CC' },
+];
+
+describe('<LicenseManager />', () => {
+  beforeEach(() => {
+    useJwt.setState({
+      jwt: 'json web token',
+    });
+
+    fetchMock.mock(
+      '/api/videos/',
+      {
+        actions: {
+          POST: {
+            license: {
+              choices: licenseChoices,
+            },
+          },
+        },
+      },
+      { method: 'OPTIONS' },
+    );
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    fetchMock.restore();
+  });
+
+  it('renders the component with existing license', async () => {
+    // Set by default with an All rights reserved license
+    const mockedVideo = videoMockFactory();
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <LicenseManager />
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    screen.getByText('License');
+    const textbox = await screen.findByRole('textbox', {
+      name: 'Select the license under which you want to publish your video, NO_CC',
+    });
+    expect(textbox).toHaveValue('All rights reserved');
+  });
+
+  it('renders the component with no license', async () => {
+    // Set by default with an All rights reserved license
+    const mockedVideo = videoMockFactory();
+    mockedVideo.license = null;
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <LicenseManager />
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    screen.getByText('License');
+    const textbox = await screen.findByRole('textbox', {
+      name: 'Select the license under which you want to publish your video',
+    });
+    expect(textbox).toHaveValue('');
+  });
+
+  it('renders the component with a license a successfuly updates it', async () => {
```

Comment:
I am not sure to understand the sentence.


Typo:
```suggestion
  it('renders the component with a license a successfully updates it', async () => {
```

Reply:
IMO, add a test as well if the update fail.

Reply:
It's "and" ahah